dist/api_data.py

```diff
@@ -1099,6 +1099,22 @@ def __ge__(self, other):
     ]),
 ]
 
+wiredtiger_open_live_restore_configuration = [
+    Config('live_restore', '', r'''Live restore configuration options. These options control the
+    behavior of WiredTiger when restoring from a backup.''', type='category', subconfig = [
+        Config('enabled', 'false', r'''whether live restore is enabled or not.''', type='boolean'),
+        Config('path', '', r'''the path to the backup that will be restored from.'''),
+        Config('threads_max', '8', r'''
+            maximum number of threads WiredTiger will start to migrate data from the backup to the
+            running WiredTiger database.''',
+            min=1, max=20),
```

Comment:
These numbers are fine. I suggest looking for `Must match` lines in `api_data.py` and have a #define for the maximum and minimum number of threads. Then you can allocate the sessions based on the maximum and you don't have to worry about consuming too many if someone reconfigures to 20.

Reply:
I don't see the need for #define if we can extract the value at setup time, I've implemented something like that here: [2f0e714](https://github.com/wiredtiger/wiredtiger/pull/11295/commits/2f0e71413e294424ed9dc4b834acda9bcdced09b)